### PR TITLE
 Fix line numbers and improve error reporting for configuration files  

### PR DESCRIPTION
**Pull request by quantumagi**
- Line numbers for the purpose of reporting errors are incorrect when blank lines are present.  The cause is the 'RemoveEmptyEntries' used to split the contents of the file into lines. This leads to those lines not being counted.
- Currently the reported line numbers start at zero. 
- There is also some repeated code present that already occurs elsewhere.

This fix:
- Changes the string split option to 'None'
- Starts line numbering at 1. Changed name of variable to reflect its usage.
- Removes any whitespace surrounding keys and values.
- Removes duplicated code by utilizing "Add"
- Adds an error case
- Adds comments

### CODE REVIEW

**Comment by dangershony** [comment]
Hi why was this closed?

**Comment by quantumagi** [comment]
Sorry, let me re-open it. Was thinking of cleaning out the reverted commit but I guess it won't have any impact.

File: Stratis.Bitcoin/Configuration/TextFileConfiguration.cs [hunk 1]
```diff
@@ -58,37 +58,37 @@ public TextFileConfiguration(Dictionary<string, List<string>> args)
...
+            {
+                // Track line numbers, also for empty lines..
+                lineNumber++;
+                var line = l.Trim();
+                // From here onwards don't process empty or commented lines
+                if (string.IsNullOrEmpty(line) || line.StartsWith("#"))
+                    continue;
+                // Split on '='
+                string[] split = line.Split('=');
+                if (split.Length == 1)
```
Comment (dangershony) [inline Stratis.Bitcoin/Configuration/TextFileConfiguration.cs]: What about 
```
if(split.Length == 0)
	continue;
```

**Comment by dangershony** [comment]
Good job I should have read your comments in more detail.

**Pull request merged**